### PR TITLE
[WIP] Json API

### PR DESCRIPTION
**Pull request by mfeckie**
So, here's what I've got done so far.

Three endpoints, counters, gauges and time_series.  The expected output is shown in the controller/README.md.

This API README is auto generated using the Bureaucrat tool if we run `DOC=1 mix test` - Not sure the current location is the right one, so looking for feedback.

I'll take a look at returning aggregate date tomorrow.

_Task List_
- [x] Counter endpoint
- [x] Gauges endpoint
- [x] Time series endpoint
- [x] Aggregate endpoint

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -338,3 +338,7 @@ I hope to improve this algorithm to be faster in the next few releases.
 ## Important Links
 
   * [Documentation](http://hexdocs.pm/beaker)
+
```
Comment (hahuang65) [inline README.md:342]: Maybe let's make a link here to `API_doc.md` so people can easily view it from Github, or do you think it'll be used internal only?

**Comment by mfeckie** [comment]
@hahuang65 I think this is ready for review / merge now.

The other things I'll put in a future PR, if you're happy with this.
- [ ] Allow query params to narrow the range of result returned
- [ ] Revise the charting JS to get JSON, rather than full page refresh

File: lib/beaker/time.ex [hunk 2]
```diff
@@ -2,8 +2,7 @@ defmodule Beaker.Time do
   @moduledoc false
 
   def now do
-    {mega, sec, micro} = :os.timestamp()
-    (mega * 1000000 + sec) * 1000000 + micro
+    :os.system_time(:micro_seconds)
```
Comment (hahuang65) [inline lib/beaker/time.ex:5]: Awesome changes in this file. Thanks for that!

Comment (mfeckie) [inline lib/beaker/time.ex:5]: Welcome, it's new stuff from Erlang 18

**Comment by hahuang65** [comment]
Very well put together PR! Really appreciate it! :)

**Comment by mfeckie** [comment]
:fireworks:  Woot!

**Comment by hahuang65** [comment]
I'm gonna hold off on releasing this until you get that next PR in, unless you'd like me to put it out. This release would have been completely your work!

**Comment by mfeckie** [comment]
I think it would be nice to get the charts PR in first, so I'm cool with that

**Comment by hahuang65** [comment]
@mfeckie really excited for this :) I was sort of just in maintenance mode on this project, but I think I'll start picking it up again. I need to make the backend a bit more solid. Love to hear any ideas you have too!

**Comment by mfeckie** [comment]
I'll take a look at the backend after I get the frontend done.  I suspect it's gonna take a few days :)

**Comment by alxndr** [comment]
:metal:

**Comment by mfeckie** [comment]
@hahuang65 Progressing with the frontend portion...

Here's a screenshot of where I've got to so far ..

screen shot 2016-01-24 at 10 41 15 pm

**Pull request merged**